File: cameo-api/src/main/java/fr/ill/ics/cameo/base/ResultEvent.java
package fr.ill.ics.cameo.base;
/*
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under the EUPL, Version 1.1 only (the "License");
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * http://joinup.ec.europa.eu/software/page/eupl
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and
 * limitations under the Licence.
 */




public class ResultEvent extends Event {
	
	private byte[] data;
	
	public ResultEvent(int id, String name, byte[] data) {
		super(id, name);
		this.data = data;
	}
	
	@Override
	public String toString() {
		return "ApplicationResult [id=" + id + ", name=" + name + "]";
	}
	
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		ResultEvent other = (ResultEvent) obj;
		
		if (id != other.id) {
			return false;
		}
		if (name == null) {
			if (other.name != null) {
				return false;
			}
		} else if (!name.equals(other.name)) {
			return false;
		}
		return true;
	}

	public byte[] getData() {
		return data;
	}
	
}

File: cpp/api/src/base/ResultEvent.cpp
/*
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under the EUPL, Version 1.1 only (the "License");
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * http://joinup.ec.europa.eu/software/page/eupl
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and
 * limitations under the Licence.
 */

#include "ResultEvent.h"
#include "JSON.h"
#include <iostream>

namespace cameo {

ResultEvent::ResultEvent(int id, const std::string& name, const std::string& data) :
	Event{id, name},
	m_data{data} {
}

ResultEvent::ResultEvent(const ResultEvent& event) :
	Event{event}, m_data{event.m_data} {
}

ResultEvent* ResultEvent::clone() {
	return new ResultEvent{*this};
}

const std::string& ResultEvent::getData() const {
	return m_data;
}

std::string ResultEvent::toString() const {
	json::StringObject jsonObject;

	jsonObject.pushKey("type");
	jsonObject.pushValue("result");

	jsonObject.pushKey("id");
	jsonObject.pushValue(m_id);

	jsonObject.pushKey("name");
	jsonObject.pushValue(m_name);

	return jsonObject.dump();
}

std::ostream& operator<<(std::ostream& os, const ResultEvent& event) {
	os << event.toString();

	return os;
}

}
